WeakAurasOptions/WeakAurasOptions.lua

```diff
@@ -72,64 +72,72 @@ function WeakAuras.MultipleDisplayTooltipDesc()
   return desc;
 end
 
-function WeakAuras.DuplicateAura(data)
-  local base_id = data.id .. " ";
-  local num = 2;
+function WeakAuras.DuplicateAura(data, newParent)
+  local base_id = data.id .. " "
+  local num = 2
 
   -- if the old id ends with a number increment the number
   local matchName, matchNumber = string.match(data.id, "^(.-)(%d*)$")
   matchNumber = tonumber(matchNumber)
   if (matchName ~= "" and matchNumber ~= nil) then
-    base_id = matchName;
+    base_id = matchName
     num = matchNumber + 1
   end
 
-  local new_id = base_id .. num;
+  local new_id = base_id .. num
   while(WeakAuras.GetData(new_id)) do
-    new_id = base_id .. num;
-    num = num + 1;
-  end
-
-  local newData = {};
-  WeakAuras.DeepCopy(data, newData);
-  newData.id = new_id;
-  newData.parent = nil;
-  newData.uid = WeakAuras.GenerateUniqueID();
-  WeakAuras.Add(newData);
-  WeakAuras.NewDisplayButton(newData);
-  if(data.parent) then
-    local parentData = WeakAuras.GetData(data.parent);
-    local index;
-    for i, childId in pairs(parentData.controlledChildren) do
-      if(childId == data.id) then
-        index = i;
-        break;
+    new_id = base_id .. num
+    num = num + 1
+  end
+
+  local newData = {}
+  WeakAuras.DeepCopy(data, newData)
+  newData.id = new_id
+  newData.parent = nil
+  newData.uid = WeakAuras.GenerateUniqueID()
+  if newData.controlledChildren then
+    newData.controlledChildren = {}
+  end
+  WeakAuras.Add(newData)
+  WeakAuras.NewDisplayButton(newData)
+  if(newParent or data.parent) then
+    local parentId = newParent or data.parent
+    local parentData = WeakAuras.GetData(parentId)
+    local index
+    if newParent then
+      index = #parentData.controlledChildren + 1
+    else
+      for i, childId in pairs(parentData.controlledChildren) do
```

Comment:
There's a utility function in Utils.lua, tIndexOf that does this, so you can replace this loop with:
index = tIndexOf(parentData.controlledChildren, data.id)